src/message.rs

```diff
@@ -69,6 +135,18 @@ impl RtMessage {
         self.tags.len() as u32
     }
 
+    pub fn tags(&self) -> &[Tag] {
+        &self.tags
+    }
+
+    pub fn values(&self) -> &[Vec<u8>] {
+        &self.values
+    }
+
+    pub fn into_hash_map(self) -> HashMap<Tag, Vec<u8>> {
```

Comment:
FYI my hidden intent in `RtMessage` was to provide a `Map` interface via O(log N) lookup on the sorted keys.

Maybe one day. 😜

Reply:
Couldn't help myself, see #2